swarm/storage/mru/topic.go

```diff
@@ -0,0 +1,56 @@
+package mru
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/ethereum/go-ethereum/common/bitutil"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/swarm/storage"
+)
+
+const topicLength = storage.KeyLength
+
+type Topic struct {
+	content [topicLength]byte
+}
+
+func NewTopic(name string, relatedContent storage.Address) Topic {
+	var topic Topic
+	copy(topic.content[:], relatedContent)
+	nameLength := len(name)
+	if nameLength > topicLength {
+		nameLength = topicLength
+	}
+	bitutil.XORBytes(topic.content[:], topic.content[:], []byte(name)[:nameLength])
```

Comment:
swarm doesn't use `bitutil`. Should we include a new package just for this?

Reply:
What would be the issue of using a new package? It is just a function I am bringing in

Reply:
never mind, I wrote this before I saw your "alchemy" further down.